packages/core/schematics/migrations/static-queries/typescript/class_declaration.ts

```diff
@@ -0,0 +1,32 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import * as ts from 'typescript';
+
+/** Determines the base type identifiers of a specified class declaration. */
+export function getBaseTypeIdentifiers(node: ts.ClassDeclaration): ts.Identifier[]|null {
+  if (!node.heritageClauses) {
+    return null;
+  }
+
+  return node.heritageClauses.filter(clause => clause.token === ts.SyntaxKind.ExtendsKeyword)
+      .reduce((types, clause) => types.concat(clause.types), [] as ts.ExpressionWithTypeArguments[])
+      .map(typeExpression => typeExpression.expression)
+      .filter(expression => ts.isIdentifier(expression)) as ts.Identifier[];
```

Comment:
nit; using the function reference instead of a new lambda will retain the `node is ts.Identifier` return type of `ts.isIdentifier(node: ts:Node)`, which in turn selects the `filter` overload that narrows the result type accordingly :)

```suggestion
      .filter(ts.isIdentifier);
```